sotodlib/site_pipeline/calibration/wiregrid.py

```diff
@@ -0,0 +1,434 @@
+# fundamental functions for the det_gamma calibration by the wire grid
+# Please contact hnakata_JP(zhongtianjiaxin@gmail.com) if you have questions.
+
+# import modules
+# python standard
+import numpy as np
+from scipy import odr
+from matplotlib.patches import Ellipse, Circle
+
+# SO specific
+from sotodlib import core
+import sotodlib.io.load_smurf as ls
+import sotodlib.hwp.hwp as hwp
+from sotodlib.hwp.g3thwp import G3tHWP
+# for house-keeping data
+from sotodlib.io import hk_utils
+from sotodlib.site_pipeline import util
+
+wg_counts2rad = 2*np.pi/52000
+#wg_degpercount = 360/52000
+
+logger = util.init_logger('wiregrid', 'wiregrid: ')
+
+# Wrap house-keeping data
+def _wrap_wg_hk(tod, telescope=None):
+    """
+    Wrap the house-keeping data about the wire grid operation.
+
+    Parameters
+    ----------
+        tod : AxisManager
+        telescope : telescope type, e.g. satp1, satp3, etc.
+            this parameter will basically be filled by the obs_info wrapped in the axismanager.
+
+    Returns
+    -------
+        tod : AxisManager
+            This includes fields, which are related with the wire grid hardware.
+            enc_rad : wires' direction read by encoder in radian
+            LSL1 : ON/OFF status of the limit switch LEFT 1 (outside) of the actuator
+            LSL2 : ON/OFF status of the limit switch LEFT 2 (inside) of the actuator
+            LSR1 : ON/OFF status of the limit switch RIGHT 1 (outside) of the actuator
+            LSR2 : ON/OFF status of the limit switch RIGHT 2 (inside) of the actuator
+
+    """
+    if telescope is None:
+        try:
+            _telescope = tod.obs_info.telescope
+        except:
+            print("Cannot load the telescope info. Please assign your target telescope")
+            logger.info("Telescope assignment was failed in site_pipeline.calibrate.wiregrid._wrap_wg_hk")
+    #
+    alias_dict = {
+        'enc_rad': _telescope + '.wg-encoder.feeds.wgencoder_full.reference_count',
+        'limitswitchL1': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL1',
+        'limitswitchL2': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSL2',
+        'limitswitchR1': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR1',
+        'limitswitchR2': _telescope + '.wg-actuator.feeds.wgactuator.limitswitch_LSR2',
+        }
+    hk_dir = '/so/level2-daq/' + _telescope + '/hk'
+    #
+    _gl_aman = hk_utils.get_detcosamp_hkaman(tod,
+                                     alias = [_i for _i in alias_dict.keys()],
+                                     fields = [_j for _j in alias_dict.values()],
+                                     data_dir = hk_dir)
+    _gl_data = {}
+    for _i in alias_dict.keys():
+        if _i == 'enc_rad':
+            _gl_data[_i] = _gl_aman['wg-encoder']['wg-encoder'][0]
+            pass
```

Comment:
Remove "pass" from non-empty blocks -- it does nothing.  I am seeing this in a lot of places...